File: app/src/main/java/com/example/dkdus/cashbook/RegistActivity.java
package com.example.dkdus.cashbook;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.example.dkdus.cashbook.databinding.ActivityRegistBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class RegistActivity extends AppCompatActivity {

    FirebaseAuth firebaseAuth;
    private String TAG = RegistActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final ActivityRegistBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_regist);

        firebaseAuth = FirebaseAuth.getInstance();

        binding.registerBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = binding.username.getText().toString();
                final String pwd = binding.password.getText().toString();

                firebaseAuth.createUserWithEmailAndPassword(email, pwd)
                        .addOnCompleteListener(RegistActivity.this, new OnCompleteListener<AuthResult>() {
                            @Override
                            public void onComplete(@NonNull Task<AuthResult> task) {
                                if(task.isSuccessful()){
                                    Intent intent = new Intent(RegistActivity.this, LoginActivity.class);
                                    startActivity(intent);
                                    finish();
                                }else if(pwd.length() < 6){
                                    Toast.makeText(RegistActivity.this, task.getException().getMessage(),Toast.LENGTH_SHORT).show();
                                }else{
                                    Toast.makeText(RegistActivity.this,"이미 아이디가 존재합니다.",Toast.LENGTH_SHORT).show();
                                }
                            }
                        });
            }
        });

    }
}

File: app/src/main/java/com/example/dkdus/cashbook/activity/RegistActivity.kt
package com.example.dkdus.cashbook.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.dkdus.cashbook.R
import com.example.dkdus.cashbook.activity.LoginActivity
import com.example.dkdus.cashbook.activity.RegistActivity
import com.example.dkdus.cashbook.databinding.ActivityRegistBinding
import com.google.firebase.auth.FirebaseAuth

class RegistActivity : AppCompatActivity() {
    var firebaseAuth: FirebaseAuth? = null
    private val TAG = RegistActivity::class.java.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityRegistBinding = DataBindingUtil.setContentView(this, R.layout.activity_regist)
        firebaseAuth = FirebaseAuth.getInstance()
        binding.registerBtn.setOnClickListener {
            val email = binding.username.text.toString()
            val pwd = binding.password.text.toString()
            firebaseAuth!!.createUserWithEmailAndPassword(email, pwd)
                    .addOnCompleteListener(this@RegistActivity) { task ->
                        if (task.isSuccessful) {
                            val intent = Intent(this@RegistActivity, LoginActivity::class.java)
                            startActivity(intent)
                            finish()
                        } else if (pwd.length < 6) {
                            Toast.makeText(this@RegistActivity, task.exception!!.message, Toast.LENGTH_SHORT).show()
                        } else {
                            Toast.makeText(this@RegistActivity, "이미 아이디가 존재합니다.", Toast.LENGTH_SHORT).show()
                        }
                    }
        }
    }
}